### DISCUSSION TITLE
term.buffer.active.getLine can't get the ansi data written by write？

### DISCUSSION THREAD

**Discussion by alanhe421** [sourcelair/xterm.js#5113] [Q&A] [answered]
Getting each line of data via term.buffer.active.getLine, is it still possible to get the Escape Sequence written by write?

For example, I'd like to go and find out this information in the terminal line header

```
\u001b]1337;PostExecMarker;Exit=0;CurrentDir=/root;\u0007
```

 I want to get the CurrentDir

When I pass `line.translateToString(false)`, I realize that there is no more Ansi data.

**Comment by jerch** [comment, marked as answer]
You cannot get the original terminal sequence from the terminal buffer, as it contains the state _after_ processing sequences.

Furthermore the sequence you want to use is not supported by xterm.js. It is a private one from iterm2. Means xterm.js will silently swallow it not processing anything. You can prolly extract the data of the sequence with preparsing the incoming data stream. Whether it is worth the trouble - idk.

**Reply by alanhe421** [reply]
I understand your meaning. Thanks.

**Reply by alanhe421** [reply]
I understand that data like the current directory  in hidden sequences. I don't know how it can find the hidden sequence corresponding to that line of directory information based on the location when a file name is selected under iterm2,

For example, in iterm2, how to do selection can let you get the selected cols and rows, but I need to know how to combine the terminal write data to find the hidden sequence of the target row.




![image](https://github.com/user-attachments/assets/128d9ca4-c5f1-48fe-bffc-6e6d0905fc65)

**Reply by jerch** [reply]
Im sorry, I dont quite understand what you want to achieve. Is the sequence sent or received by iterm2? Does it figure out automatically the CWD on a mouse selection over a file listing in the viewport, and would send that sequence to app side? Idk what this is about sorry.

**Reply by alanhe421** [reply]
Anyway, thank you.
